src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/json/ReadJSONStep.java

```diff
@@ -0,0 +1,184 @@
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import hudson.Extension;
+import net.sf.json.JSON;
+
+/**
+ * Reads a JSON file from the workspace.
+ *
+ * @author Nikolas Falco
+ */
+public class ReadJSONStep extends AbstractStepImpl {
+
+    private String file;
+    private String text;
+
+    @DataBoundConstructor
+    public ReadJSONStep() {
+    }
+
+    /**
+     * The path to a file in the workspace to read JSON content from.
+     *
+     * @return the file path
+     */
+    public String getFile() {
+        return file;
+    }
+
+    /**
+     * The path to a file in the workspace to read JSON content from.
+     *
+     * @param file the path to file in the workspace
+     */
+    @DataBoundSetter
+    public void setFile(String file) {
+        this.file = file;
+    }
+
+    /**
+     * A String containing JSON formatted data.
+     *
+     * @return text to parse
+     */
+    public String getText() {
+        return text;
+    }
+
+    /**
+     * A String containing JSON formatted data.
+     *
+     * @param text to parse
+     */
+    @DataBoundSetter
+    public void setText(String text) {
+        this.text = text;
+    }
+
+    @Extension
+    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+
+        public DescriptorImpl() {
+            super(ReadJSONStepExecution.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "readJSON";
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Read JSON from files in the workspace.";
+        }
+
+        @Override
+        public Step newInstance(Map<String, Object> arguments) throws Exception {
+            ReadJSONStep step = new ReadJSONStep();
+            if (arguments.containsKey("file")) {
+                Object file = arguments.get("file");
+                if (file != null) {
+                    step.setFile(file.toString());
+                }
+            }
+            if (arguments.containsKey("text")) {
+                Object text = arguments.get("text");
+                if (text != null) {
+                    step.setText(text.toString());
+                }
+            }
+            if (isNotBlank(step.getFile()) && isNotBlank(step.getText())) {
+                throw new IllegalArgumentException("At most one of file or text must be provided to " + getFunctionName() + '.');
+            }
+            if (isBlank(step.getFile()) && isBlank(step.getText())) {
+                throw new IllegalArgumentException("At least one of file or text needs to be provided to " + getFunctionName() + '.');
+            }
+            return step;
+        }
+    }
+
+    @Extension
+    public static class JSONWhiteLister extends Whitelist {
```

Comment:
This whitelist only permit the basic isArray and size methods needed to navidate json object in pipeline. The json implementation I decide to use (instead GJSON or similar) is because is shipped by default with Jenkins.
In script-security-plugin I see only JVM or groovy library packages so is the right place for **net.sf.json**, for example the step readMavenPOM do the same. Are you moving all custom whitelist to that plugin?